Source/Core/VideoBackends/OGL/ProgramShaderCache.cpp

```diff
@@ -182,161 +220,214 @@ void ProgramShaderCache::UploadConstants()
   }
 }
 
-SHADER* ProgramShaderCache::SetShader(u32 primitive_type)
+SHADER* ProgramShaderCache::SetShader(u32 primitive_type, const GLVertexFormat* vertex_format)
```

Comment:
This function feels spiritually very similar to the D3D one (haven't looked at Vulkan yet).  Feels like we might be able to move some of this to VideoCommon in the future.